cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -183,13 +186,32 @@ public final ProgramController run(final Program program, final ProgramOptions o
       final ProgramOptions options = addArtifactPluginFiles(oldOptions, localizeResources,
                                                             DirUtils.createTempDir(tempDir));
 
+      final List<String> additionalClassPaths = new ArrayList<>();
+      // Add extra jars set in cConf to additionalClassPaths and localizeResources
+      for (URI jarURI : CConfigurationUtil.getExtraJars(cConf)) {
+        LocalizeResource localizeResource;
+        String protocol = jarURI.getScheme();
+        if (protocol == null) {
+          localizeResource = new LocalizeResource(new File(jarURI.getPath()));
+        } else if ("file".equals(protocol)) {
+          localizeResource = new LocalizeResource(new File(jarURI));
+        } else {
+          localizeResource = new LocalizeResource(jarURI, false);
+        }
+        String jarName = new File(jarURI.getPath()).getName();
+        localizeResources.put(jarName, localizeResource);
+        String jarPath = "hdfs".equals(protocol) ? jarURI.toString() : jarName;
```

Comment:
This seems wrong to be used as classpath (`jarURI.toString()`). Also, as mentioned above, shouldn't check for "hdfs" explicitly.